### ISSUE TITLE
[API] Add a way to specify whether an expansion can be used from an async context

### ISSUE THREAD

**Issue by iGabyTM** [PlaceholderAPI/PlaceholderAPI#895]
### Confirmation

- [X] I checked the Issues and Pull request tab for any existing issues or PRs.
- [X] My Feature Request is for **PlaceholderAPI** and not any expansion or other plugin.

### Type

New API feature, Major Code improvement (Will affect Servers)

### Description

See https://discord.com/channels/164280494874165248/1040979624039022602

**Comment by Andre601** [comment]
Could you also share the full context?

The link doesn't work on mobile and people may not share the server this is in.

**Comment by ancap-kun** [comment]
No, all placeholders have to be async

**Comment by Andre601** [comment]
> No, all placeholders have to be async

I don't think that would be the right aproach. Async-only placeholder handling could cause issues in certain situations, especially those where Spigot will complain that something can only be executed sync/async.

**Comment by spikehidden** [comment]
> Could you also share the full context?
> 
> The link doesn't work on mobile and people may not share the server this is in.

May I add this:
<details>
<summary>Screenshot</summary>

![](https://i.alexgoodwin.media/i/misc/2fdd06.png)

</details>

This is actually a good and urgent reason to provide a way to fetch if the placeholders are safe to be called async. It probably never happens anything but then the whole server breaks.

As an alternative all plugins/extensions that are not safe for pulling data async should block such requests (like Denizen) but I doubt that this will happen.

**Greetings**!
![Lurking Fox](https://github.com/user-attachments/assets/3e03ecf3-d5e6-4712-8a82-0202bb4558b9)

**Comment by Andre601** [comment]
Something I would like to mention is, that we're open for suggestions, or even Merge requests, that help with making the expansion and placeholder system more thread-safe.

I myself do not have any real knowledge of ways that could make the current system more thread safe, but if mcmonkey has some suggestions is he free to share them with us, so that we can try and implement them. Tho, preferably would we like actual Merge requests to be made, as our free time is very limited, so any contributions to get this working more safely would be apreciated.

Personally speaking, I feel like expansions should remain async by default, but perhaps receive a method to override (i.e. `boolean isSyncOnly()`) to tell PlaceholderAPI that this expansion may only be run in a synchronous way, so that it would then handle it differently. How is again something that would need discussion, which we're very open to have here.
